task5/src/main/java/com/clouway/adapter/http/BankOperationHandler.java

```diff
@@ -35,10 +38,17 @@ public BankOperationHandler(FundsRepository fundsRepository, Validator validator
   protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
     Cookie[] cookies = request.getCookies();
     Cookie cookie = cookieFinder.find(cookies);
+    if (cookie == null) {
+      response.sendRedirect("/login?errorMsg=Session expired! Please log in again!");
```

Comment:
This check has to be in the security filter. If there is no cookie client should not reach here. If you need it her to fix bug this means there is something wrong with the security workflow. Need to be fixed